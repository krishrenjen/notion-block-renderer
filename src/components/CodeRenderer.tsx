import { FC } from "react";
import { annotationToClassName } from "../utils";
import { Context } from "../utils";
import { useContext } from "react";
import { CodeProps, RichTextType } from "../types";
import SyntaxHighlighter from "react-syntax-highlighter";
import {
  monokaiSublime,
  irBlack,
  tomorrowNightBright,
  monokai,
} from "react-syntax-highlighter/dist/cjs/styles/hljs";

const CodeRenderer: FC<CodeProps> = ({ lang, richTextArr }) => {
  const { prefix, isCodeHighlighter } = useContext(Context);

  if (isCodeHighlighter) {
    console.log(44445555, isCodeHighlighter);
    return (
      <div className={`language-${lang}`}>
        <SyntaxHighlighter
          language={lang}
          style={tomorrowNightBright}
          className="rounded-lg"
          customStyle={{ padding: "1rem" }}
          // showLineNumbers={true}
        >
          {getJoinedText(richTextArr)}
        </SyntaxHighlighter>
      </div>
    );
  }

  return (
    <pre>
      <code className={`language-${lang}`}>
        {richTextArr.map((richText: any, index: number) => {
          const className = annotationToClassName(richText.annotations, prefix);
          return (
            <span key={index} className={className}>
              {richText.text.content}
            </span>
          );
        })}
      </code>
    </pre>
  );
};

export default CodeRenderer;

/**
 *
 * @param richTextArr
 * @returns
 */
const getJoinedText = (richTextArr: RichTextType[]): string => {
  const textArr = richTextArr.map(
    (richText: any, index: number) => richText.text.content
  );
  return textArr.join("");
};
