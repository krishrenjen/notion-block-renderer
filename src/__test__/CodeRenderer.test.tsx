import { create } from "react-test-renderer";
import CodeRenderer from "../components/CodeRenderer";
import { createContext } from "react";
import { useContext } from "react";
import {
  BLOCKS_PREFIX,
  BLOCK_PREFIX,
  IS_CODE_HIGHLIGHTER,
  IS_NEXTJS,
  PREFIX,
  SYNTAX_HIGHLIGHTER_CSS,
} from "../config";

const annotations = {
  bold: false,
  code: false,
  italic: false,
  strikethrough: false,
  underline: false,
  color: "default",
};

const richTextArr = [
  {
    plain_text: `export enum Property {
      Name = "Name",
      cover = "cover",
      Author = "Author",
      Description = "Description",
      Slug = "Slug",
      Public = "Public",
      Published = "Published",
      Tags = "Tags",
    }`,
    href: "",
    annotations: annotations,
    type: "",
    //   text: {
    //     content: string;
    //     link: {
    //       url: string;
    //     };
    //   };
  },
];

export const Context = createContext({
  prefix: PREFIX,
  isNextJS: IS_NEXTJS,
  blockPrefix: BLOCK_PREFIX,
  blocksPrefix: BLOCKS_PREFIX,
  // isCodeHighlighter: IS_CODE_HIGHLIGHTER,
  isCodeHighlighter: true,
  syntaxHighlighterCSS: SYNTAX_HIGHLIGHTER_CSS,
});

it("changes the class when hovered", () => {
  const component = create(
    <Context.Provider
      value={{
        prefix: PREFIX,
        blockPrefix: BLOCK_PREFIX,
        blocksPrefix: BLOCKS_PREFIX,
        isNextJS: IS_NEXTJS,
        // isCodeHighlighter: IS_CODE_HIGHLIGHTER,
        isCodeHighlighter: true,
        syntaxHighlighterCSS: SYNTAX_HIGHLIGHTER_CSS,
      }}
    >
      <CodeRenderer lang="typescript" richTextArr={richTextArr} />
    </Context.Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseEnter();
  // });
  // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseLeave();
  // });
  // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
