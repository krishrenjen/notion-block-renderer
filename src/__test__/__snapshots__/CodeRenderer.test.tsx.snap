// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<pre>
  <code
    className="language-typescript"
  >
    <span
      className=""
    >
      export enum Property {
      Name = "Name",
      cover = "cover",
      Author = "Author",
      Description = "Description",
      Slug = "Slug",
      Public = "Public",
      Published = "Published",
      Tags = "Tags",
    }
    </span>
  </code>
</pre>
`;
